src/app/login/page.tsx

```diff
@@ -0,0 +1,59 @@
+"use client";
+
+import useUserInfoStore from "@/hooks/useUserInfoStore";
+import { useRouter } from "next/navigation";
+
+const LoginPage = () => {
+  const router = useRouter();
+
+  const onSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    const input = e.currentTarget.querySelector("input");
+    if (!input) {
+      return;
+    }
+
+    const userId = input.value;
+    if (!userId) {
+      return;
+    }
+
+    // 로그인 처리
+    const response = await fetch(`/api/mock-auth`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({ userId }),
+    });
```

Comment:
api폴더에 함수로 빼서 처리하면 관심사 분리 측면에서 더 나아질 것 같긴 한데 나중에 작업하셔도 무방할 것 같습니다!

Reply:
body에 userId를 담는 것 말씀이신가요? api에 POST요청 시 body에 담아야 api에서 받을 수 있는 걸로 알았는데, 따로 분리할 수 있는 방법이 있을까요??

Reply:
같이 문서작업하느라 작업 리뷰가 늦었네요ㅎㅎ
제가 말씀드렸던건 

**api 폴더 내에 함수 작성 후 export**

```suggestion
export const SubmitUserId = (userId) => {
   const response = await fetch(`/api/mock-auth`, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({ userId }),
    });
    
    
    if (!response.ok) {
      throw new Error("로그인에 실패했습니다.");
    }

    const userInfo = await response.json();
    useUserInfoStore.setState({
      _id: userInfo._id,
      name: userInfo.name,
      color: userInfo.color,
      hostingRooms: userInfo.hostingRooms,
      joinedRooms: userInfo.joinedRooms,
      token: userInfo.token,
    });

    window.localStorage.setItem("token", userInfo.token);
    router.push(`/dashboard/${userInfo._id}`);
  };
    
}
```


login/page.tsx에서는 다음과 같이사용하여 가독성 개선 및 관심사 분리 가능
```suggestion
import SubmitUserId from "~/api"

Submit(userId)
```

이 의도로 말씀드린 것이었습니다! 
생각보다 함수가 긴데 중간에 끼워넣어서 오해가 있을법 하게 말했네요ㅠㅠ